lib/icinga/apiactions.cpp

```diff
@@ -555,5 +556,164 @@ Dictionary::Ptr ApiActions::GenerateTicket(const ConfigObject::Ptr&,
+		if (!CheckCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(400, "Command '" + resolved_command + "' is not of type '" + command_type + "'.");
+	} else if (command_type == "EventCommand") {
+		if (!EventCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(400, "Command '" + resolved_command + "' is not of type '" + command_type + "'.");
+	} else if (command_type == "NotificationCommand") {
+		if (!NotificationCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(400, "Command '" + resolved_command + "' is not of type '" + command_type + "'.");
+	}
+
+	/* Get TTL param */
+	if (!params->Contains("ttl"))
+		return ApiActions::CreateResult(400, "Parameter ttl is required.");
+
+	double ttl = HttpUtility::GetLastParameter(params, "ttl");
+	if (ttl <= 0)
+		return ApiActions::CreateResult(400, "Parameter ttl must be greater than 0.");
+
+	/* This generates a UUID */
+	String uuid = Utility::NewUniqueID();
+
+	/* Create the deadline */
+	double deadline = Utility::GetTime() + ttl;
+
+	/* Update executions */
+	Dictionary::Ptr pending_execution = new Dictionary();
+	pending_execution->Set("pending", true);
+	pending_execution->Set("deadline", deadline);
+	Dictionary::Ptr executions = checkable->GetExecutions();
+	if (!executions)
+		executions = new Dictionary();
+	executions->Set(uuid, pending_execution);
+	checkable->SetExecutions(executions);
+
+	/* Create execution parameters */
+	Dictionary::Ptr execParams = new Dictionary();
+	execParams->Set("command_type", command_type);
+	execParams->Set("command", resolved_command);
+	execParams->Set("host", host->GetName());
+	if (service)
+		execParams->Set("service", service->GetShortName());
+
+	/*
+	 * If the host/service object specifies the 'check_timeout' attribute,
+	 * forward this to the remote endpoint to limit the command execution time.
+	 */
+	if (!checkable->GetCheckTimeout().IsEmpty())
+		execParams->Set("check_timeout", checkable->GetCheckTimeout());
+
+	execParams->Set("source", uuid);
+	execParams->Set("deadline", deadline);
+
+	bool local = endpointPtr == Endpoint::GetLocalEndpoint();
+	if (local) {
+		MessageOrigin::Ptr origin = new MessageOrigin();
+		ClusterEvents::ExecuteCommandAPIHandler(origin, execParams);
+	} else {
+		/* Broadcast the update */
+		ApiListener::Ptr listener = ApiListener::GetInstance();
+		if (!listener)
+			return ApiActions::CreateResult(404, "No ApiListener instance available.");
+
+		Dictionary::Ptr updateParams = new Dictionary();
+		updateParams->Set("host", host->GetName());
+		if (service)
+			updateParams->Set("service", service->GetShortName());
+		updateParams->Set("executions", executions);
+
+		Dictionary::Ptr updateMessage = new Dictionary();
+		updateMessage->Set("jsonrpc", "2.0");
+		updateMessage->Set("method", "event::UpdateExecutions");
+		updateMessage->Set("params", updateParams);
+
+		listener->SyncSendMessage(endpointPtr, updateMessage);
```

Comment:
Ah! Overseen that this is `event::UpdateExecutions`. Here (in opposite to `event::ExecuteCommand`) you need `RelayMessage`.